patches/chromium/expose_a_method_for_invalidating_internal_cache.patch

```diff
@@ -0,0 +1,43 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: "emmanuel.kimmerlin@thomsonreuters.com"
+ <emmanuel.kimmerlin@thomsonreuters.com>
+Date: Mon, 18 Nov 2019 12:42:23 +0100
+Subject: Expose a method for invalidating internal cache
+
+Allow to customize Electron/Chromium cache location
+See MR: https://github.com/electron/electron/pull/19693
+
+diff --git a/base/path_service.cc b/base/path_service.cc
+index d1c9bb1045a5d4cff7ded79ebced53ce4c2daa3c..01ef3fd68bd1683b973a9545639c3d2ad116fae3 100644
+--- a/base/path_service.cc
++++ b/base/path_service.cc
+@@ -344,6 +344,15 @@ void PathService::RegisterProvider(ProviderFunc func, int key_start,
+   path_data->providers = p;
+ }
+ 
++// static
++void PathService::InvalidateCache() {
++  PathData * path_data = GetPathData();
++  DCHECK(path_data);
++
++  AutoLock scoped_lock(path_data->lock);
++  path_data->cache.clear();
++}
++
+ // static
+ void PathService::DisableCache() {
+   PathData* path_data = GetPathData();
+diff --git a/base/path_service.h b/base/path_service.h
+index 14779f16481c1c6592471b771f9547316551d39c..f4ca101cea460f5eef9313fd94eccfb868d10cac 100644
+--- a/base/path_service.h
++++ b/base/path_service.h
+@@ -78,6 +78,9 @@ class BASE_EXPORT PathService {
+   // Disable internal cache.
+   static void DisableCache();
+ 
++  // Clear cache
++  static void InvalidateCache();
++
+  private:
+   friend class ScopedPathOverride;
+   FRIEND_TEST_ALL_PREFIXES(PathServiceTest, RemoveOverride);
```

Comment:
Just leaving a breadcrumb: we should try to upstream this patch. @emmkimme do you want to do that?

I'm still :+1: on this approach of having upstream have an "invalidate cache" API so that we can use this PathService code rather than reinventing the wheel with a custom PathService in Electron's code.

Reply:
I never did it.
But to be fair, according to the time this MR is pending could we bypass this step please, would be so happy to have it in master as soon as possible.